### PR TITLE
Add PyArrow integration test for C Stream Interface

### LINKED ISSUE #1847: PyArrow intergation test for C Stream Interface
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

We added C Stream Interface previously but haven't add the PyArrow integration test as C Data Interface. We should add the test.

**Describe the solution you'd like**

Add PyArrow intergation test for C Stream Interface

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #1847.

# Rationale for this change
 
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by viirya** [comment]
cc @sunchao

File: arrow/src/ffi_stream.rs [hunk 1]
```diff
@@ -198,13 +198,6 @@ impl ExportedArrayStream {
     }
 
     pub fn get_schema(&mut self, out: *mut FFI_ArrowSchema) -> i32 {
-        unsafe {
-            match (*out).release {
-                None => (),
-                Some(release) => release(out),
-            };
-        };
-
```
Comment (viirya) [inline arrow/src/ffi_stream.rs:207]: For `get_schema` and `get_next`, we did it wrongly to releasing the output struct. It should be handled by the caller.

**Review by sunchao — APPROVED** [review]
LGTM. Thanks @viirya !

**Comment by viirya** [comment]
Thanks @sunchao !

**Pull request merged**